pkg/lint/lintersdb/manager.go

```diff
@@ -39,8 +40,24 @@ func (m *Manager) WithCustomLinters() *Manager {
 		m.log = report.NewLogWrapper(logutils.NewStderrLog(""), &report.Data{})
 	}
 	if m.cfg != nil {
+		var basePath string
+		var err error
+		if m.cfg.ConfigFilePath != nil {
+			basePath, err = filepath.Abs(filepath.Dir(*m.cfg.ConfigFilePath))
+			if err != nil {
+				m.log.Errorf("Unable to get base directory based on config: %s. %v",
+					*m.cfg.ConfigFilePath, err)
+			}
+		} else {
+			basePath, err = os.Getwd()
+			if err != nil {
+				m.log.Errorf("Unable to get current working directory. %v",
+					err)
+			}
+		}
+
```

Comment:
The other way we could do this is resolve the `settings.Path` in the config reader. It would mean we don't need to add the `ConfigFilePath` property and carry it around, but I thought it might be nice to know the origin of the config for future cases where we might want to "anchor" relative paths to the config file.